userCode.json

```diff
@@ -0,0 +1 @@
connection.setDoOutput(true);","        DataOutputStream out = new DataOutputStream(connection.getOutputStream());","        out.writeBytes(ParameterStringBuilder.getParamsString(parameters));","        out.flush();","        out.close();","","","        if (connection.getResponseCode() >= 300) {","            Reader streamReader = new InputStreamReader(connection.getErrorStream());","            System.out.println(streamReader.read());","        } else {","            try (InputStream in = connection.getInputStream()) {","                Path target = Paths.get(\"\", parameters.get(\"name\") + \".zip\").toAbsolutePath();","                Files.copy(in, target, StandardCopyOption.REPLACE_EXISTING);","                return parameters.get(\"name\") + \".zip\";","            }","        }","        return null;","    }","","","    public void unzipFile(String in, String target) throws IOException {","        File destDir = new File(target);","","        byte[] buffer = new byte[1024];","        ZipInputStream zis = new ZipInputStream(new FileInputStream(in));","        ZipEntry zipEntry = zis.getNextEntry();","        while (zipEntry != null) {","            File newFile = newFileFromZip(destDir, zipEntry);","            if (zipEntry.isDirectory()) {","                if (!newFile.isDirectory() && !newFile.mkdirs()) {","                    throw new IOException(\"Failed to create directory \" + newFile);","                }","            } else {","                // fix for Windows-created archives","                File parent = newFile.getParentFile();","                if (!parent.isDirectory() && !parent.mkdirs()) {","                    throw new IOException(\"Failed to create directory \" + parent);","                }","","                // write file content","                FileOutputStream fos = new FileOutputStream(newFile);","                int len;","                while ((len = zis.read(buffer)) > 0) {","                    fos.write(buffer, 0, len);","                }","                fos.close();","            }","            zipEntry = zis.getNextEntry();","        }","","        zis.closeEntry();","        zis.close();","    }","","    private File newFileFromZip(File destinationDir, ZipEntry zipEntry) throws IOException {","        File destFile = new File(destinationDir, zipEntry.getName());","","        String destDirPath = destinationDir.getCanonicalPath();","        String destFilePath = destFile.getCanonicalPath();","","        if (!destFilePath.startsWith(destDirPath + File.separator)) {","            throw new IOException(\"Entry is outside of the target dir: \" + zipEntry.getName());","        }","","        return destFile;","    }","","    public boolean newDirectoryFromPath(String destinationDir, String newDirectory) {","        File theDir = new File(destinationDir + newDirectory);","        if (!theDir.exists()) {","            return theDir.mkdirs();","        }","        return false;","    }","","}",""]}]}
```

Comment:
Kann man die beim Erzeugen auch schön formatiert anlegen? Wäre vielleicht nett fürs debugging. Wenn es zu viel Aufwand ist können wir es auch so lassen. Vielleicht gibt es aber auch irgendeine Methode in der Library dafür, don't know.